### PR TITLE
SRX-7ZA5ML Update build and publish commands for kuberpult

### PR DESCRIPTION
**Pull request by mnishamk-freiheit**
- Update build and publish commands such that they will build the images and publish them
- Build and publish images with kuberpult-builder when pull request is created
- Use version number from `git describe` to avoid collisions

Verified that images are created for the version
- [europe-west3-docker.pkg.dev/fdc-public-docker-registry/kuberpult/kuberpult-builder:0.4.17-23-g2d946b5](https://console.cloud.google.com/artifacts/docker/fdc-public-docker-registry/europe-west3/kuberpult/kuberpult-builder/sha256:8b89ca7117e3b54696ff0a4ade2b4805868e4a42b098fc806cac7115bbabb536)
- https://console.cloud.google.com/artifacts/docker/fdc-public-docker-registry/europe-west3/kuberpult/kuberpult-cd-service/sha256:870e918696b10a7e0a9451a4fb233ca7a80e1ffdb5b29dec17b9455c60abf642?project=fdc-public-docker-registry%5C
- https://console.cloud.google.com/artifacts/docker/fdc-public-docker-registry/europe-west3/kuberpult/kuberpult-frontend-service/sha256:db989e8c3d9365f30a5e301d939d3f9247f2efb8d24ae4cd75eef13dd7bf8a40?project=fdc-public-docker-registry%5C

### CODE REVIEW

File: .github/workflows/execution-plan-snippet.yml [hunk 1]
```diff
@@ -41,15 +41,21 @@ jobs:
       - name: Set version from tag
         if: inputs.version != ''
         run: |
-          echo "VERSION=${{ inputs.version }}" >> $GITHUB_ENV
+          echo "${{ inputs.version }}" > version
       - name: Set version automatically 
         if: inputs.version == ''
         run: |
-          echo "VERSION=$(git describe --always --long --tags || echo 0.0.1)" >> $GITHUB_ENV
+          echo "$(git describe --always --long --tags || echo 0.0.1)" > version
```
Comment (fdcds) [inline .github/workflows/execution-plan-snippet.yml:48]:
```suggestion
          (git describe --always --long --tags || echo 0.0.1) > version
```

File: .github/workflows/execution-plan-snippet.yml [hunk 2]
```diff
@@ -88,12 +94,15 @@ jobs:
       options: --user root
       credentials:
         username: "_json_key"
-        password: ${{ secrets.FDC_CORE_CI_IMAGE_READER }}
+        password: ${{ secrets.GCP_ARTIFACT_REGISTRY_PUSH_JSON_KEY }}
```
Comment (fdcds) [inline .github/workflows/execution-plan-snippet.yml:97]: This is needed to be able to pull the `kuberpult-builder` image?

File: services/frontend-service/Makefile [hunk 3]
```diff
@@ -14,8 +14,10 @@
 #along with kuberpult.  If not, see <http://www.gnu.org/licenses/>.
 
 #Copyright 2021 freiheit.com
-MAKEFLAGS += --no-builtin-rules
 
+include ../../Makefile.variables
+
+MAKEFLAGS += --no-builtin-rules
```
Comment (fdcds) [inline services/frontend-service/Makefile:20]: Why is this needed?

File: services/frontend-service/Makefile [hunk 4]
```diff
@@ -44,13 +46,16 @@ run: bin/main
...
-build-pr: build
+build-pr: docker
 	echo "build on pull request"
 
-build-main: build
+build-main: docker
 	echo "build on main"
 
-publish: 
+login:
```
Comment (fdcds) [inline services/frontend-service/Makefile:55]: This should probably be marked `.PHONY`?

File: services/cd-service/Makefile [hunk 5]
```diff
@@ -95,7 +98,9 @@ release: docker
 
 all: test docker
 
-publish:
-	echo "nothing to do"
+login:
```
Comment (fdcds) [inline services/cd-service/Makefile:101]: This should probably be marked `.PHONY`?

Comment (mnishamk-freiheit) [reply services/cd-service/Makefile:101]: is publish is never created as a file, would phony matter?

File: services/frontend-service/Makefile (hunk 3 above)
Comment (mnishamk-freiheit) [reply services/frontend-service/Makefile:20]: makeflags already existed.
the makefile variable was added to get IMAGE_REGISTRY variable

File: .github/workflows/execution-plan-snippet.yml (hunk 2 above)
Comment (mnishamk-freiheit) [reply .github/workflows/execution-plan-snippet.yml]: removed it. public image so not required now

File: infrastructure/docker/builder/Makefile [hunk 6]
```diff
@@ -1,4 +1,4 @@
-VERSION=0.0.1
+VERSION=$(shell cat ../../../version)
```
[the remark this answers is not in the document]
Comment (fdcds) [reply infrastructure/docker/builder/Makefile:1]: In a conversation with @mnishamk-freiheit we decided to postpone this to a later story.

File: services/cd-service/Makefile [hunk 7]
```diff
@@ -15,14 +15,17 @@
...
+
 MAKEFLAGS += --no-builtin-rules
 
 VERSION=$(shell cat ../../version)
 export VERSION
 
 export CGO_ENABLED=1
 
-IMAGENAME?=ghcr.io/freiheit-com/kuberpult-cd-service:${VERSION}
+IMAGENAME?=$(IMAGE_REGISTRY)/kuberpult-cd-service:${VERSION}
```
Comment (fdcds) [inline services/cd-service/Makefile:28]:
```suggestion
IMAGENAME?=$(IMAGE_REGISTRY)/kuberpult-cd-service:$(VERSION)
```

**Review by fdcds — APPROVED** [review]
